libs/libGamedata/gameData/SelectionMapInputData.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+#include <boost/filesystem.hpp>
+#include <string>
+#include <vector>
+
+struct ImageResource
+{
+    boost::filesystem::path filePath;
+    unsigned index;
+    ImageResource(boost::filesystem::path path = boost::filesystem::path(), unsigned index = 0)
+        : filePath(std::move(path)), index(index){};
+};
+
+struct MissionSelectionInfo
+{
+    unsigned maskAreaColor = 0;
+    Position ankerPos;
+};
+
+struct SelectionMapInputData
+{
+    SelectionMapInputData() = default;
+
+    // background image for the selection map
+    ImageResource background;
+    // the map image itself. Pixels which should be transparent must have the value of transparentColor
```

Comment:
In general this could also be a single image. But in the original settler 2 we have to different images for this. So i decided to following this approach. And with a separate background image you have also the posibility to place the selection map in middle of the background and have a border around

Reply:
>  But in the original settler 2 we have to different images for this.

Ok so there we need it

> And with a separate background image you have also the posibility to place the selection map in middle of the background and have a border around

For this you could create one image of the background and the map in the middle, i.e. the combination of background and map as intended in a single image. Or why does this not work?

Reply:
Yes your are right but for the original world selection screen i have to create a new image out of the both to do this. So i decided to stay at the approach with to different images

Reply:
And i am not sure if it is allowed to ship this combined image for legal reasons.